dotCMS/src/main/java/com/dotcms/saml/DotAbstractSamlConfigurationServiceImpl.java

```diff
@@ -0,0 +1,254 @@
+
+        try {
+
+            final String[] array = identityProviderConfiguration.containsOptionalProperty(samlName.getPropertyName())?
+                    StringUtils.split((String) identityProviderConfiguration.getOptionalProperty(samlName.getPropertyName()), DotSamlConstants.ARRAY_SEPARATOR_CHAR):
+                    this.getDefaultArrayStringParameter(samlName);
+
+            Logger.debug(this, ()-> "Found " + samlName.getPropertyName() + " : " + ((array == null) ? "null" : array));
+
+            return array;
+        } catch (Exception e) {
+
+            Logger.warn(this, "Cast exception on " + samlName.getPropertyName()
+                    + " property. idpConfigId: " + identityProviderConfiguration.getId());
+        }
+
+        return null;
+    }
+
+    @Override
+    public Integer getConfigAsInteger(final IdentityProviderConfiguration identityProviderConfiguration, final SamlName samlName) {
+
+        try {
+
+            final Integer value = identityProviderConfiguration.containsOptionalProperty(samlName.getPropertyName())?
+                Integer.parseInt((String) identityProviderConfiguration.getOptionalProperty(samlName.getPropertyName())):
+                this.getDefaultIntegerParameter(samlName);
+
+            Logger.debug(this, ()-> "Found " + samlName.getPropertyName() + " : " + ((value == null) ? "null" : value));
+
+            return value;
+        } catch (Exception e) {
+
+            Logger.warn(this, "Cast exception on " + samlName.getPropertyName()
+                    + " property. idpConfigId: " + identityProviderConfiguration.getId());
+        }
+
+        return null;
+    }
+
+    ////////
+
+    private Integer getDefaultIntegerParameter(final SamlName samlName) {
+
+        if (samlName == null) {
+
+            throw new SamlException("The 'getDefaultIntegerParameter' property is null");
+        }
+
+        if (this.defaultProperties.containsKey(samlName.getPropertyName())) {
+
+            return Integer.parseInt(this.defaultProperties.get(samlName.getPropertyName()));
+        }
+
+        throw new SamlException(NOT_FOUND_ERROR + samlName.getPropertyName());
+    }
+
+    private String[] getDefaultArrayStringParameter(final SamlName samlName) {
+
+        final String value = this.getDefaultStringParameter(samlName);
+        if (value != null) {
+            return StringUtils.split(value, DotSamlConstants.ARRAY_SEPARATOR_CHAR);
+        }
+
+        throw new SamlException(NOT_FOUND_ERROR + samlName.getPropertyName());
+    }
+
+    public String getDefaultStringParameter(final SamlName property) {
+
+        if (property == null) {
+
+            throw new SamlException("The 'getDefaultStringParameter' property is null");
+        }
+
+        if (this.defaultProperties.containsKey(property.getPropertyName())) {
+
+            final String string = this.defaultProperties.get(property.getPropertyName());
+            return NULL == string? null : string;
```

Comment:
Codacy found an issue: [Use equals() to compare object references.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5707914)